Split windows.py into a folder with separate modules

mcdo0486 opened the pull request:
Splits windows.py into a folder with separate modules.
This will make it easier to develop new window classes outside of a monolithic module.

msmttchr commented:
> Splits windows.py into a folder with separate modules. This will make it easier to develop new window classes outside of a monolithic module.

Thanks again for your effort and contribution.
I think it's best to wait outcome of #594 before reviewing this, since the principle is the same. Just a small comment for the time being, do you think it's worth splitting `ManagedImageWindow` in a separate file?

msmttchr reviewed (approved):
Ok for me. Possibly @CasperSchippers can double check.

mcdo0486 commented:
I fixed some documentation for widgets in a [598](https://github.com/pymeasure/pymeasure/pull/598). I'll do the same here once that has been merged because some links will need to be updated.

CasperSchippers commented on pymeasure/display/windows/image_window.py:
```diff
@@ -0,0 +1,65 @@
+log.addHandler(logging.NullHandler())
+
+
+class ManagedImageWindow(ManagedWindow):
```
I'd be inclined to name this file managed_image_window.py, seems a bit more clear/consistent to me.

mcdo0486 replied:
Good point, I renamed the file.

CasperSchippers commented on pymeasure/display/windows/managed_window.py:
```diff
@@ -630,7 +537,8 @@ def directory(self, value):
 
 class ManagedWindow(ManagedWindowBase):
```
Just a thought; would it not also make sense to split this file into one for ManagedWindowsBase and one for ManagedWindow?

bilderbuchi replied:
I'd say multiple classes per module is definitely fine in python, and imo having every class in a separate module does not improve the code structure.

CasperSchippers reviewed (commented):
Left a few comments, mostly looks good to me.

mcdo0486 commented:
> I fixed some documentation for widgets in a [598](https://github.com/pymeasure/pymeasure/pull/598). I'll do the same here once that has been merged because some links will need to be updated.

Thanks for merging the other pull request. I've updated the various widget docstrings to be consistent with the refactor in this pull request.

mcdo0486 commented on pymeasure/display/windows/managed_window.py:
```diff
@@ -1,16 +1,3 @@
-# copies of the Software, and to permit persons to whom the Software is
-# furnished to do so, subject to the following conditions:
-#
-# The above copyright notice and this permission notice shall be included in
```
I fixed that section.

bilderbuchi commented:
@CasperSchippers good to merge?

CasperSchippers reviewed (approved):
Yes, looks good to me.
Thanks for the work and sorry for the long wait.
